containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -1,116 +1,291 @@
+
+        if hasattr(self, "data") and self.valid:
+            json_file = json.dumps(self.data)
             s3_client.put_object(
-                Body=json_metadata,
-                Bucket=self.metadata_bucket,
-                Key=self.metadata_key,
+                Body=json_file,
+                Bucket=self.write_bucket,
+                Key=self.write_key,
                 **{
                     "ACL": "bucket-owner-full-control",
                     "ServerSideEncryption": "AES256",
                 },
             )
-            self.logger.info("Data Product metadata written to s3")
+            self.logger.info(f"Data Product {self.type} written to s3")
         else:
-            self.logger.error(
-                "Metadata need to be validated before writing to s3, run the validate() class method"
-            )
+            self.logger.error(f"{self.type} is not valid to write.")
             raise ValidationError(
                 "Metadata must be validated before being written to s3."
             )
+
+    # compare old and new metadata/schema and use
+    # logic to set version scale.
+    # will need new path function to reflect new version number schema and metadatas 3 paths
+    def _classify_update(self):
+        """infer type of increment to version, major/minor"""
+        pass
+
+    # this can be passed to a path function for schema and metadata that doesn't use latest version
+    # Maybe this will be better suited outside of the class.
+    def _generate_new_version_number(self):
+        """generate a version number to pass to get path function"""
+        pass
+
+
+class DataProductSchema(DataProductBaseJsonSchema):
+    """
+    class to handle creation and updating of
+    schema json files relating to data product tables
+    """
+
+    def __init__(
+        self, data_product_name: str, table_name: str, logger: DataPlatformLogger
+    ):
+        bucket_path = BucketPath.from_uri(
+            DataProductConfig(name=data_product_name).schema_path(table_name).uri
+        )
+        self.table_name = table_name
+
+        super().__init__(
+            data_product_name, logger, JsonSchemaName("schema"), bucket_path
+        )
+
+        if not self._does_data_product_metadata_exist():
+            self.logger.error("Data product metadata not yet registered.")
+            self.has_registered_data_product = False
+        else:
+            self.has_registered_data_product = True
+
+    def _does_data_product_metadata_exist(self):
+        """checks wheter data product for schema has metadata registered"""
+        data_product_metadata_path = BucketPath.from_uri(
+            DataProductConfig(name=self.data_product_name).metadata_path().uri
+        )
+        md_bucket = data_product_metadata_path.bucket
+        md_key = data_product_metadata_path.key
+        return self._check_if_metadata_or_schema_exists(
+            md_bucket, md_key, JsonSchemaName("metadata")
+        )
```

Comment:
alternatively, if every `DataProductBaseJsonSchema` instance has a `check_if_exists()` method:

```suggestion
        metadata = DataProductMetadata(self.data_product_name, self.logger)
        return metadata.check_if_exists()
```